packages/next/src/server/dev/hot-reloader-webpack.ts

```diff
@@ -1091,11 +1091,13 @@ export default class HotReloaderWebpack implements NextJsHotReloaderInterface {
     this.multiCompiler.hooks.done.tap('NextjsHotReloader', () => {
       inputFileSystem.purge!()
     })
-    watchCompilers(
-      this.multiCompiler.compilers[0],
-      this.multiCompiler.compilers[1],
-      this.multiCompiler.compilers[2]
-    )
+    if (this.config.logging !== false) {
+      watchCompilers(
+        this.multiCompiler.compilers[0],
+        this.multiCompiler.compilers[1],
+        this.multiCompiler.compilers[2]
+      )
+    }
```

Comment:
@huozhi Instead of passing config value down to `build/output/store.ts`, conditionally run `watchCompilers` here, lmk if you don't like this approach!

P.S. similar approach for `hot-reloader-turbopack` above.

Reply:
Can we introduce a state property for logging into output store? Then we can set the value in hot reloader and use that state to determine if we need to log anything inside the store

Reply:
Yes, but the reason for this is that we have to drill down the value multiple times by `buildStore.setState`, so I made it conditional above.

Would it still be better to pass the value?

Reply:
I think it will be more clear that `logging` state only controling the logging behavior instead of controlling the whole compiler watching. If you need to change sth later in watchCompilers for other purpose rather than logging, it would be difficult to refactor